torch/_dynamo/variables/tensor.py

```diff
@@ -471,13 +473,49 @@ def call_method(
         from . import ConstantVariable, TupleVariable
 
         kwargs = dict(kwargs)
-
+        # print("CALLING TENSOR OP", name)
         options = VariableTracker.propagate(self, args, kwargs.values())
         if name == "stride" and self.stride is not None:
             constant_result = ConstantVariable(self.stride, **options)
         elif name == "size" and self.size is not None:
             sizes = [variables.ConstantVariable(x) for x in self.size]
             constant_result = SizeVariable(sizes, **options)
+        elif name == "size" and self.size is None and config.dynamic_shapes:
+            example_size = self.proxy.node.meta["example_value"].size()
+            pure_size_proxy = tx.output.create_proxy(
+                "call_method",
+                name,
+                *proxy_args_kwargs([self], kwargs),
+                current_tx=tx,
+            )
+            if len(args) == 1 and isinstance(args[0], ConstantVariable):
+                assert isinstance(args[0].value, int)
+                proxy = tx.output.create_proxy(
+                    "call_function",
+                    operator.getitem,
+                    (pure_size_proxy, args[0].as_proxy()),
+                    {},
```

Comment:
What's the policy for desugaring into the graph? Technically, you didn't call size()[0], the user asked for size(0). This can matter for things like NestedTensor, for which the size is only well defined on some dimensions (and variable on others). It seems better to more faithfully represent what originally happened here.